CHANGELOG.md

```diff
@@ -11,6 +11,8 @@ For experimental package changes, see the [experimental CHANGELOG](experimental/
 
 ### :rocket: (Enhancement)
 
+perf(sdk-trace-base): do not allocate arrays if resource has no pending async attributes
```

Comment:
nit. makes it easier when I have to cut the next release :slightly_smiling_face: 

```suggestion
* perf(sdk-trace-base): do not allocate arrays if resource has no pending async attributes
```